contracts/implementations/nodes/SyntheticDevice.sol

```diff
@@ -166,6 +248,62 @@ contract SyntheticDevice is AccessControlInternal {
         );
     }
 
+    /**
+     * @notice Burns a synthetic device and reset all its attributes and links
+     * @dev Caller must have the admin role
+     * @dev This contract has the BURNER_ROLE in the SyntheticDeviceId
+     * @param vehicleNode Vehicle node id
+     * @param syntheticDeviceNode Synthetic Device node id
+     * @param ownerSig Vehicle/Synthetic Device's owner signature hash
+     */
+    function burnSyntheticDeviceSign(
+        uint256 vehicleNode,
+        uint256 syntheticDeviceNode,
+        bytes calldata ownerSig
+    ) external onlyRole(Roles.DEFAULT_ADMIN_ROLE) {
+        NodesStorage.Storage storage ns = NodesStorage.getStorage();
+        MapperStorage.Storage storage ms = MapperStorage.getStorage();
+        SyntheticDeviceStorage.Storage storage sds = SyntheticDeviceStorage
+            .getStorage();
+
+        address vehicleIdProxyAddress = VehicleStorage
+            .getStorage()
+            .idProxyAddress;
+        address sdIdProxyAddress = sds.idProxyAddress;
+
+        if (
+            ms.nodeLinks[vehicleIdProxyAddress][sdIdProxyAddress][
+                vehicleNode
+            ] != syntheticDeviceNode
+        ) revert Errors.VehicleNotPaired(vehicleNode);
+
+        address owner = INFT(sdIdProxyAddress).ownerOf(syntheticDeviceNode);
```

Comment:
Note that this is the first place where we would catch `vehicleNode = syntheticDeviceNode = 0` or simply `syntheticDeviceNode = 0` and the vehicle being unpaired; this line exploding.

Reply:
That's a catch